pkgs/applications/science/electronics/kicad/base.nix

```
@@ -15,58 +15,77 @@
 , boost
 , pkg-config
 , doxygen
+, graphviz
 , pcre
 , libpthreadstubs
 , libXdmcp
 , lndir
 
+, util-linux
+, libselinux
+, libsepol
+, libthai
+, libdatrie
+, libxkbcommon
+, epoxy
+, dbus
+, at-spi2-core
+, libXtst
+
+, swig
+, python
+, wxPython
+, opencascade
+, opencascade-occt
+, libngspice
+, valgrind
+
 , stable
 , baseName
 , kicadSrc
 , kicadVersion
 , i18n
 , withOCE
-, opencascade
 , withOCC
-, opencascade-occt
 , withNgspice
-, libngspice
 , withScripting
-, swig
-, python
-, wxPython
 , debug
-, valgrind
+, sanitize
 , withI18n
-, gtk3
+, withWayland
 }:
 
 assert lib.asserts.assertMsg (!(withOCE && stdenv.isAarch64)) "OCE fails a test on Aarch64";
 assert lib.asserts.assertMsg (!(withOCC && withOCE))
   "Only one of OCC and OCE may be enabled";
 let
-  inherit (lib) optional optionals;
+  inherit (lib) optional optionals optionalString;
 in
 stdenv.mkDerivation rec {
   pname = "kicad-base";
-  version = kicadVersion;
+  version = if (stable) then kicadVersion else builtins.substring 0 10 src.rev;
 
   src = kicadSrc;
 
   # tagged releases don't have "unknown"
   # kicad nightlies use git describe --dirty
   # nix removes .git, so its approximated here
-  # "-1" appended to indicate we're adding a patch
   postPatch = ''
     substituteInPlace CMakeModules/KiCadVersion.cmake \
-      --replace "unknown" "${builtins.substring 0 10 src.rev}-1" \
-      --replace "${version}" "${version}-1"
+      --replace "unknown" "${builtins.substring 0 10 src.rev}"
+  ''
+  + optionalString (withWayland) ''
+    substituteInPlace kicad/kicad.cpp \
+      --replace "x11" "wayland"
+    substituteInPlace common/single_top.cpp \
+      --replace "x11" "wayland"
+    substituteInPlace  libs/kiplatform/gtk/environment.cpp \
+      --replace "x11" "wayland"
```

This could be simplified.

```suggestion
    for f in kicad/kicad.cpp common/single_top.cpp libs/kiplatform/gtk/environment.cpp; do
      substituteInPlace $f --replace "x11" "wayland"
    done
```